sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -505,7 +505,7 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     val schema = StructType(Seq(
       StructField("f", StructType(Seq(
         StructField("a", StringType, nullable = true),
-        StructField("b", IntegerType, nullable = false)
+        StructField("b", IntegerType, nullable = true)
```

Comment:
ok. so the new test (`row nullability mismatch`) is effectively covering this case? Then, should we change the name of this test? Will we hit the exception that is checked by this test in any other cases?